kubevirt_scripts/mountcd.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+#### IMPORTANT: This script is only meant to show how to implement required scripts to make custom hardware compatible with FakeFish. Dell hardware is supported by the `idrac-virtualmedia` provider in Metal3.
+#### This script has to mount the iso in the server's virtualmedia and return 0 if operation succeeded, 1 otherwise
+#### Note: Iso image to mount will be received as the first argument ($1)
+#### You will get the following vars as environment vars
+#### BMC_ENDPOINT - Has the BMC IP
+#### BMC_USERNAME - Has the username configured in the BMH/InstallConfig and that is used to access BMC_ENDPOINT
+#### BMC_PASSWORD - Has the password configured in the BMH/InstallConfig and that is used to access BMC_ENDPOINT
+
+ISO=${1}
+CLUSTER_STORAGE_CLASS=ocs-storagecluster-ceph-rbd
+IS_HTTPS=false
+
+if [ $(echo ${ISO} | grep -c https://) -eq 1 ]
+then
+  HOSTNAME=$(echo ${ISO} | awk -F "https://" '{print $2}' | awk -F "/" '{print $1}' | awk -F ":" '{print $1}')
+  PORT=$(echo ${ISO} | awk -F "https://" '{print $2}' | awk -F "/" '{print $1}' | awk -F ":" '{print $2}')
+  if [ -z ${PORT} ]
+  then
+    PORT=443
+  fi
+  openssl s_client -showcerts -connect ${HOSTNAME}:${PORT} </dev/null 2>/dev/null | openssl x509 -outform PEM > /tmp/iso-endpoint-ca.crt
+  IS_HTTPS=true
+fi
+
+export VM_NAME=$(echo $BMC_ENDPOINT | awk -F "_" '{print $1}')
+export VM_NAMESPACE=$(echo $BMC_ENDPOINT | awk -F "_" '{print $2}')
+
+export KUBECONFIG=/var/tmp/kubeconfig
+
+# we need to poweroff the VM if it's running
+VM_RUNNING=$(oc -n ${VM_NAMESPACE} get vm ${VM_NAME} -o jsonpath='{.spec.running}')
+VM_WAS_RUNNING=false
+if [[ "${VM_RUNNING}" == "true" ]]
+then
+  VM_WAS_RUNNING="true"
+  virtctl -n ${VM_NAMESPACE} stop ${VM_NAME}
+fi
+
+if [ ${IS_HTTPS} == "true" ]
+then
+  oc -n ${VM_NAMESPACE} delete configmap ${VM_NAME}-iso-ca &> /dev/null
+  oc -n ${VM_NAMESPACE} create configmap ${VM_NAME}-iso-ca --from-file=ca.crt=/tmp/iso-endpoint-ca.crt
+
+  cat <<EOF | oc apply -f -
+  apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    annotations:
+      cdi.kubevirt.io/storage.import.certConfigMap: "${VM_NAME}-iso-ca"
+      cdi.kubevirt.io/storage.import.endpoint: "${ISO}"
+      cdi.kubevirt.io/storage.bind.immediate.requested: "true"
+    name: ${VM_NAME}-bootiso
+    namespace: ${VM_NAMESPACE}
+  spec:
```

Comment:
nit: the spec part is common, move to a variable?

Reply:
done.